packages/Interface/src/DTK_C_API.cpp

```diff
@@ -0,0 +1,369 @@
+/****************************************************************************
+ * Copyright (c) 2012-2017 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ ****************************************************************************/
+
+#include "DTK_C_API.h"
+
+#include "DTK_Struct.hpp"
+#include "DTK_Version.hpp"
+
+namespace DataTransferKit
+{
+
+struct DTK_FunctionWrap
+{
+    DTK_FunctionWrap( void ( *f )(), void *data )
+        : _f( f )
+        , _data( data )
+    {
+    }
+    void ( *_f )();
+    void *_data;
+};
+
+class IntBool
+{
+  public:
+    IntBool( bool &b )
+        : _b( b )
+    {
+    }
+    ~IntBool() { _b = ( _i != 0 ); }
+    int *data() { return &_i; }
```

Comment:
I think it would be good to add a comment to explain that we need `IntBool` because C doesn't have boolean. I would also prefer if we had a function to explicitly update `_b`.